File: frontend/src/microservices/tour-group/objects/TourGroupMessageAttachmentResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupMessageAttachmentResponse`，作为响应数据并提供 JSON 编解码。

export type TourGroupMessageAttachmentResponse = {
  attachmentId: string
  attachmentType: string
  publicUrl: string
  originalFileName: string
  mimeType: string
  fileSize: number
}
export const tourGroupMessageAttachmentResponseFromJson = (json: string): TourGroupMessageAttachmentResponse =>
  JSON.parse(json) as TourGroupMessageAttachmentResponse

export const tourGroupMessageAttachmentResponseToJson = (value: TourGroupMessageAttachmentResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupMessageAttachmentResponse.scala
// TourGroupChatPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import com.typesafe.travel.shared.kernel.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

import java.time.Instant

final case class TourGroupMessageAttachmentResponse(
    attachmentId: String,
    attachmentType: String,
    publicUrl: String,
    storagePath: String,
    originalFileName: String,
    mimeType: String,
    fileSize: Long,
    sortOrder: Int,
    createdAt: String
)
object TourGroupMessageAttachmentResponse:
  given sourceEncoder: Encoder[TourGroupMessageAttachmentResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupMessageAttachmentResponse] = deriveDecoder
